README.md

```diff
@@ -1,4 +1,274 @@
+
+	```scala
+		case q"$a.take($b, $c)" if a.tpe =:= typeOf[Query[_]]
+  	 		&& b.tpe =:= typeOf[Int]
+  	 		&& c.tpe =:= typeOf[Boolean] => Take(a, b, c)
+	```
+- **no dependence with types and count of arguments**
+	- as shown in the previous example, a new argument implies a new code
+- **no verbosity**
+	- this kind of code can become illegible
+- **no duplication of code**
+	- again in the override example, there is unnecessary code duplication
+
+
+## Persisted
+why persisted
+
+## Inline
+why inline
+
+
+# Project Structure
+| *Component*                                   		  | *Description*                        | 
+|:---------                                   		  |:-----------                        | 
+| `directembedding/...` <br> `/DirectEmbedding.scala` | **DirectEmbedding code:** reification code with macro | 
+| `dsls/main/.../BasicSpec.scala`             		  | **Test:** Intermediate Representation        | 
+| `dsls/test/.../TestBase.scala`              		  | **Test:** Corners cases tests                | 
+
+## DirectEmbedding.scala
+
+This file contains the reification code. It consists of the definition of:
+
+* reifyAs()
+
+	```scala
+	   class reifyAs(to: Any) extends scala.annotation.StaticAnnotation
+	```
+	
+	* This defines the annotation *reifyAs* which accepts *Any* object so the IR can be attach into the symbols
+
+* and lift()
+
+	```scala
+		def lift[T](c: Context)(block: c.Expr[T]): c.Expr[T] = {
+   			import c.universe._
+		    class LiftingTransformer extends Transformer {
+		    	...
+		    }
+		}
+	```
+	
+	* This is the method that encompasses all the reification process. 
+
+The class LiftingTransformer defines:
+
+* reify()
+
+	```scala
+		def reify(methodSym: Symbol, targs: List[Tree],
+			args: List[Tree]): Tree = { ... }
+	```
+
+	* This code uses the macro with its parameters to reify the captured function into its IR.
+	
+* and transform()
+
+	```scala
+		override def transform(tree: Tree): Tree = { ... }
+	```
+
+	* transform() *pattern matches* over the different ASTs to extract the essential data for reify() that is to say the symbol, the arguments and the type arguments.
+
+## TestBase.scala
+This file contains the functions and IR that represent a DSL. It is used to for testing purpose in BasiSpec.scala
+
+Example for a class:
+
+```scala
+case object ClassCons extends Exp[ClassExample] // Note: IR extends the real returned type
+
+@reifyAs(ClassCons)
+class ClassExample {
+  val dummyVal: Int = 1
+}
+```
+
+Example for a function with *many* arguments:
+
+```scala
+case class AppManyArgs[T](self: Exp[TArgClassExample[T]], p1: Exp[T]*) extends Exp[T] // Note: all function have as first argument a self
+
+@reifyAs(TArgClassExampleCase)
+class TArgClassExample[T] {
+	  @reifyAs(AppManyArgs)
+	  def app1[T](p1: T*): T = ???
+}
+```
+
+## BasicSpec.scala
+This fils contains the tests.
+
+Below the test for take():
+
+```scala
+"lift" should "work with TArgClassExample methods with take" in {
```

Comment:
Give meaningful names to functions. TArgClassExample is not good. Do this in all the places.